### PR TITLE
qa-tests: add a snapshot-test workflow

### PR DESCRIPTION
**Pull request by mriccobene**

### CODE REVIEW

File: .github/workflows/snapshot-test.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+            rsync -a --delete $ERIGON_DATA_DIR/snapshots/ $ERIGON_SNAPSHOT_TEST_DIR/
+
+      - name: Run Execution Test
+        id: test_step
+        working-directory: ${{runner.workspace}}/silkworm/build/cmd
+        run: |
+          set +e # Disable exit on error
+          
+          # Run the test script
+          some_test_cmd
```
Comment (canepat) [inline .github/workflows/snapshot-test.yml:55]: We can start running some subcommands of `snapshots` utility as tests:
```
./snapshots --snapshot_dir $ERIGON_SNAPSHOT_TEST_DIR count_headers
./snapshots --snapshot_dir $ERIGON_SNAPSHOT_TEST_DIR count_bodies
```

**Review by canepat — CHANGES_REQUESTED** [review]

File: .github/workflows/snapshot-test.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
+name: QA - Snapshot Test
+
+on:
+  workflow_dispatch:
+
+jobs:
+  execution-test-suite:
+    runs-on: [ self-hosted, Erigon3 ] # must run on E2 not to steal E3 runner, which is used for PR approval
```
[the remark this answers is not in the document]
Comment (mriccobene) [reply .github/workflows/snapshot-test.yml:8]: yes

File: .github/workflows/snapshot-test.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    timeout-minutes: 7200  # 5 days
+    env:
+      ERIGON_QA_PATH: /opt/erigon-qa
+      ERIGON_DATA_DIR: /opt/erigon-versions/reference-version/datadir
+      ERIGON_SNAPSHOT_TEST_DIR: /opt/erigon-snapshot
+      CHAIN: mainnet
+      TRACKING_TIME_SECONDS: 431400  # 5 days minus 10 minutes
+      TOTAL_TIME_SECONDS: 431400
```
[the remark this answers is not in the document]
Comment (mriccobene) [reply .github/workflows/snapshot-test.yml:16]: ok

File: .github/workflows/snapshot-test.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        run: cmake -E make_directory ${{runner.workspace}}/silkworm/build
+
+      - name: Configure CMake
+        working-directory: ${{runner.workspace}}/silkworm/build
+        run: |
+          cmake $GITHUB_WORKSPACE -DCMAKE_BUILD_TYPE=Release
+
+      - name: Build Silkworm
+        working-directory: ${{runner.workspace}}/silkworm/build
+        run: cmake --build . --config Release --target silkworm -j 8
```
[the remark this answers is not in the document]
Comment (mriccobene) [reply .github/workflows/snapshot-test.yml:38]: correct

File: .github/workflows/snapshot-test.yml (hunk 1 above)
Comment (mriccobene) [reply .github/workflows/snapshot-test.yml:55]: ok

**Review by canepat — APPROVED** [review]

**Pull request merged**